frontend/src/container/JoinWorkspace.jsx

```diff
@@ -0,0 +1,181 @@
+import { getWorkspaceSubscriptions, joinWorkspace, subscribeToWorkspace } from '../action-creator.async.js'
+
+export class JoinWorkspace extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = { filter: () => true }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  async componentDidMount () {
+    this.handleAllAppChangeLanguage()
+    await this.loadWorkspaceSubscriptions()
+  }
+
+  async loadWorkspaceSubscriptions () {
+    const { props } = this
+
+    const fetchSubscriptionList = await props.dispatch(getWorkspaceSubscriptions(props.user.userId))
+
+    if (fetchSubscriptionList.status !== 200) {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+      return
+    }
+    props.dispatch(setWorkspaceSubscriptionList(fetchSubscriptionList.json))
+  }
+
+  handleAllAppChangeLanguage () {
+    const { props } = this
+    this.buildBreadcrumbs()
+    props.dispatch(setHeadTitle(props.t('Join a space')))
+  }
+
+  buildBreadcrumbs () {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: <Link to={PAGE.HOME}><i className='fa fa-home' />{props.t('Home')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }, {
+      link: <Link to={PAGE.JOIN_WORKSPACE}>{props.t('Join a space')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }]))
+  }
+
+  async joinWorkspace (workspaceId) {
+    const { props } = this
+    await props.dispatch(joinWorkspace(workspaceId, props.user.userId))
+    props.history.push(PAGE.WORKSPACE.DASHBOARD(workspaceId))
+  }
+
+  createRequestComponent (workspace) {
+    const { props } = this
+
+    const subscription = props.workspaceSubscriptionList.find(s => s.workspace.workspace_id === workspace.id)
+
+    if (subscription !== undefined) {
+      var text = 'Unknown request state'
+      var icon = 'question'
+      switch (subscription.state) {
+        case SUBSCRIPTION_TYPE.pending.slug:
+          text = props.t('Request sent')
+          icon = SUBSCRIPTION_TYPE.pending.faIcon
+          break
+        case SUBSCRIPTION_TYPE.rejected.slug:
+          text = props.t('Request rejected')
+          icon = SUBSCRIPTION_TYPE.rejected.faIcon
+          break
+      }
+      return <div><i class={`fa fa-${icon}`} /> {text}</div>
+    }
+
+    switch (workspace.accessType) {
+      case SPACE_TYPE.onRequest.slug:
+        return (
+          <IconButton
+            icon='share'
+            text={props.t('Request access')}
+            onClick={() => props.dispatch(subscribeToWorkspace(workspace.id, props.user.userId))}
+          />)
+      case SPACE_TYPE.open.slug:
+        return (
+          <IconButton
+            icon='sign-in'
+            text={props.t('Join the space')}
+            onClick={() => this.joinWorkspace(workspace.id)}
+          />)
+      default:
+        return 'Unknown space access type'
```

Comment:
if your function usually returns a component, I think it would be better if your default state would also return a "component like" thing.
Wrapping the string in `<span>` or `<div>` should be enough

Reply:
or possibly the empty element `<>..</>`?

Reply:
this is called React Fragment and its purpose is to avoid adding an extra node to the DOM.
So I don't think it will fit the needs here

Reply:
I know what it is. Why?